pkgs/applications/graphics/kcc/default.nix

```diff
@@ -1,39 +1,48 @@
-{ lib
-, mkDerivationWith
-, python3Packages
-, fetchPypi
-, p7zip
-, archiveSupport ? true
-}:
-
-mkDerivationWith python3Packages.buildPythonApplication rec {
+{ stdenv, lib, qt6, python3Packages, fetchFromGitHub, p7zip
+, archiveSupport ? true }:
+python3Packages.buildPythonApplication rec {
   pname = "kcc";
-  version = "5.5.1";
+  version = "6.2.2";
 
-  src = fetchPypi {
-    inherit version;
-    pname = "KindleComicConverter";
-    sha256 = "5dbee5dc5ee06a07316ae5ebaf21ffa1970094dbae5985ad735e2807ef112644";
+  src = fetchFromGitHub {
+    owner = "ciromattia";
+    repo = "kcc";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GX0WPu1EZx/8H1ajivxgnym4jECmwPBJU25vzcfOQCw=";
   };
 
-  propagatedBuildInputs = with python3Packages ; [
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
+  buildInputs = lib.optional stdenv.hostPlatform.isLinux qt6.qtwayland;
+  propagatedBuildInputs = (with python3Packages; [
+    packaging
     pillow
-    pyqt5
     psutil
     python-slugify
     raven
-  ];
+    requests
+    natsort
+    mozjpeg_lossless_optimization
+    distro
+    pyside6
+  ]);
 
-  qtWrapperArgs = lib.optionals archiveSupport [ "--prefix" "PATH" ":" "${ lib.makeBinPath [ p7zip ] }" ];
+  qtWrapperArgs = lib.optionals archiveSupport [
+    "--prefix"
+    "PATH"
+    ":"
+    "${lib.makeBinPath [ p7zip ]}"
+  ];
 
   postFixup = ''
     wrapProgram $out/bin/kcc "''${qtWrapperArgs[@]}"
   '';
 
   meta = with lib; {
-    description = "Python app to convert comic/manga files or folders to EPUB, Panel View MOBI or E-Ink optimized CBZ";
+    description =
+      "Python app to convert comic/manga files or folders to EPUB, Panel View MOBI or E-Ink optimized CBZ";
     homepage = "https://kcc.iosphe.re";
     license = licenses.isc;
-    maintainers = with maintainers; [ dawidsowa ];
+    maintainers = with maintainers; [ dawidsowa adfaure ];
```

Comment:
Should probably add `mainProgram = "kcc"` to the `meta` fields here.

Reply:
Oh, and should add the `changelog field too: `changelog = "https://github.com/ciromattia/kcc/releases/tag/v${version}";`